langchain/chains/router/base.py

```diff
@@ -0,0 +1,71 @@
+"""Base classes for chain routing."""
+from __future__ import annotations
+
+from abc import ABC
+from typing import Any, Dict, List, Mapping, Optional
+
+from pydantic import Extra
+
+from langchain.callbacks.manager import CallbackManagerForChainRun
+from langchain.chains.base import Chain
+
+
+class RouterChain(Chain, ABC):
+    """Chain that outputs the name of a destination chain and the inputs to it."""
+
+    @property
+    def output_keys(self) -> List[str]:
+        return ["destination", "next_inputs"]
+
+
+class MultiRouteChain(Chain):
+    """Use a single chain to route an input to one of multiple candidate chains."""
+
+    router_chain: RouterChain
+    """Chain that routes inputs to destination chains."""
+    destination_chains: Mapping[str, Chain]
+    """Chains that return final answer to inputs."""
+    default_chain: Chain
```

Comment:
should this just be a key in destination chains? or worth having entirely separate?

Reply:
imo worth having separate in case you want graceful error handling (could also store default_key and keep it in destination_chains but this was just my personal pref)